### PR TITLE
Lambda: DT configuration via environment variables, etc. 

### PR DESCRIPTION
**Pull request by tippmar-nr**
* Adds support for taking distributed trace configuration variables that normally are supplied in the connect response from either local config or environment variables when in serverless mode. 
* Updates unit tests for default config
* Cleanup for a few things in `HandlerMethodWrapper` that weren't conforming to the spec, added a couple of TODOs as placeholders until I can figure out where to get the values that are needed.
*

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs [hunk 1]
```diff
@@ -77,9 +77,13 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
             EventTypes.TryGetValue(eventTypeName, out var eventType); // handle case where the name might not be in the eventType dictionary
 
             attributes.AddEventSourceAttribute("eventType", eventType ?? "Unknown"); // TODO: Is this correct?
-            attributes.Add("aws.RequestId", requestIdGetter(lambdaContext));
+            attributes.AddEventSourceAttribute("arn", "????"); // TODO: how to get this value? Spec says "ARN of the invocation source"
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs:80]: Should we hold off on adding this until we figure out how to get it?

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs:1]: Do we have a story to remind us about the TODO regarding removing the code that logs the attributes?

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs (hunk 1 above)
Comment (tippmar-nr) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs:80]: That's really more of a placeholder so I could PR the changes I've made so far. I'll continue refining things as we move along.

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs
Comment (tippmar-nr) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs:1]: Yeah, there's a "Final code pass" story to catch any remaining TODOs.

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs (hunk 1 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs:80]: That's fine, I just want to ensure that we don't forget to do that.

**Review by nrcventura — APPROVED** [review]

**Pull request merged**